services/mailservers/tcp_ping.go

```diff
@@ -0,0 +1,41 @@
+package mailservers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/status-im/status-go/tcpping"
+)
+
+type PingQuery struct {
+	Addresses []string `json:"addresses"`
+	TimeoutMs int      `json:"timeoutMs"`
+}
+
+type PingResult struct {
+	Address string  `json:"address"`
+	Latency *int    `json:"latency"`
+	PingErr *string `json:"error"`
```

Comment:
This is totally ok. I was mistaken, having `PingErr error` will always return `"Error": {}` when marshalled to JSON unless you define a custom error type like this: https://play.golang.org/p/qlZEjqORxlO